File: 08/fall/solutions/vote_verifier.java
import java.util.*;

class vote_verifier {

	public static void main(String[] arguments) {

		Scanner in = new Scanner(System.in);

		int n = in.nextInt();
		int[] candidate = new int[n];

		while (in.hasNextInt()) {

			int index = in.nextInt();
			int cipher = in.nextInt();
			int key = in.nextInt();
			String plain = "";

			for (int i = n; --i >= 0;) {
				candidate[i] = ((cipher % 10) + (key % 10)) % 10;
				plain = candidate[i] + plain;
				cipher /= 10;
				key /= 10;
			}
			System.out.printf("plaintext ballot: %s  index: %d  vote for candidate: %d%n", plain, index, candidate[index-1]);
		}
	}
}

File: 08/fall/extra/golf/vote_verifier.py
def add(b, k):
    return ''.join(str((x+y)%10) for (x,y) in zip(map(int, b), map(int, k)))

raw_input() #first line isn't used
while True:
    try: index, ballot, key = raw_input().split()
    except: break
    plain = add(ballot, key)
    print "plaintext ballot: %s  index: %s  vote for candidate: %s" %\
            (plain, index, plain[int(index)-1])
